### PR TITLE
Add transpiler to produce pip requirements file

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
this adds a transpiler that can create a pip requirements file from a release matrix file (and a repo).

could for instance be used in workflows as input for checking dependency conflicts, checking for upgradable packages, or scanning for vulnerabilities.

### CODE REVIEW

File: tests/test_release_transpiler.py
Comment (DanSava) [inline tests/test_release_transpiler.py:60]: This is an interesting way of writing:
```python
rhel_ver = "rhel7"
py_ver = "py38"
```

File: komodo/snyk_reporting.py
Comment (valentin-krasontovitsch) [reply komodo/snyk_reporting.py:129]: oh thank you that was me debugging a snyk failure - gonna remove those :sweat_smile:

Comment (valentin-krasontovitsch) [reply komodo/snyk_reporting.py:129]: ... and all the other print statements ...

File: tests/test_release_transpiler.py
Comment (valentin-krasontovitsch) [reply tests/test_release_transpiler.py:60]: indeed! i think it comes from a time when there were several potential versions of OS and python. if i scrap it here, i'd probably have to scrap it many places in the tests and/or code... what do you think? should i keep it consistent, do lots of changes, ... ?

**Comment by kwinkunks** [comment]
Wanted to ask if you've had a chance to look at #315 to see if/how they interact?

Question: is a "matrix file" a single release file for one 'element' in the matrix, or is it the big 'all platforms, all versions' file that `combine` produces? (The latter is what I think of as a matrix file, but the language we currently use in the repo is confusing / ambiguous).

Wondering if instead of adding a new subcommand, it could be clearer to add 'filtering' to the existing `transpile` command? That is, "give me the transpiled release file(s) but only for those packages with such-and-such a build system?

Lastly, I think the new functions need docstrings.

**Comment by valentin-krasontovitsch** [comment]
thanks for bringing that issue / PR to my attention - i have not looked at it, but at first quick glance it seems orthogonal to what i try to achieve here, but would benefit from / have to be adjusted to the changes there.

i am so sorry, but i'm afraid i am not able to relieve you of your confusion, because i hold and cherish that same confusion myself : D
i suspect that the language is confusing because it doesn't accurately depict the current situation, and i try not to think about it too much, which of course works as long as i don't need to talk to anybody about details ^^

thing is, the transpilation as it stands does not produce "what i want", namely a pip requirements file of the form
```
pkg1==version1
pkg2==version2
```
rather i think it produces just some different yaml.

docstrings, yes, of course, i will look into that, thank you : )

File: tests/test_release_transpiler.py
Comment (berland) [reply tests/test_release_transpiler.py:60]: It is only a matter of time before we add py310 ...

Comment (valentin-krasontovitsch) [reply tests/test_release_transpiler.py:60]: great! and i guess we will support both 3.8 and 3.10, at least for a while?

**Comment by xjules** [comment]
> thanks for bringing that issue / PR to my attention - i have not looked at it, but at first quick glance it seems orthogonal to what i try to achieve here, but would benefit from / have to be adjusted to the changes there.
> 
> i am so sorry, but i'm afraid i am not able to relieve you of your confusion, because i hold and cherish that same confusion myself : D i suspect that the language is confusing because it doesn't accurately depict the current situation, and i try not to think about it too much, which of course works as long as i don't need to talk to anybody about details ^^
> 
> thing is, the transpilation as it stands does not produce "what i want", namely a pip requirements file of the form
> 
> ```
> pkg1==version1
> pkg2==version2
> ```
> 
> rather i think it produces just some different yaml.
> 
> docstrings, yes, of course, i will look into that, thank you : )

I guess regardless of this PR we should try to merge [this](https://github.com/equinor/komodo/pull/315) first, where in the additional parameters (python and os versions) can be linked into your PR or?

**Comment by valentin-krasontovitsch** [comment]
yes totally! gonna wait with this for now, and rebase / adjust once the other transpiler PR is merged

**Comment by valentin-krasontovitsch** [comment]
rebasing is done, can i please get another review to get his merged? @yngve-sk

**Review by yngve-sk — APPROVED** [review]
LGTM🚀

**Pull request merged**
